### PR TITLE
[FEATURE] Introduce Site Settings Editor

### PR DESCRIPTION
**Pull request by linawolf**
Related: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/1026
Releases: main

### CODE REVIEW

File: Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst [hunk 1]
```diff
@@ -8,88 +8,187 @@ Site settings definitions
 =========================
 
 ..  versionadded:: 13.1
+    Site-scoped setting definitions where introduced. They will most likely be
+    the place to configure site-wide configuration, which was previously only
+    possible to modify via Constant Editor, modifying TypoScript constants.
```
Comment (froemken) [inline Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst:13]: I don't understand that part "modifying via Constants Editor" and "modifying TypoScript constants"

File: Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst [hunk 2]
```diff
@@ -8,88 +8,187 @@ Site settings definitions
...
 
-The big problem with TypoScript and TSConfig is that each specified value can
-only ever be a string. It is up to the developer alone to read these values
-and convert them into the desired data type such as integer.
+Site settings definitions allow to define settings with a type and a guaranteed
+default value. They can be defined in :ref:`site-sets`, in a file called
+:file:`_blog_settins.definitions.yaml`.
```
Comment (froemken) [inline Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst:17]: The file is wrong. It's `settings` and without `blog` and without `_` in front.

File: Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst [hunk 3]
```diff
@@ -8,88 +8,187 @@ Site settings definitions
...
+
+        ..  confval:: type
+            :type: a :ref:`definition type <definition-types>`
+            :name: site-settings-definition-settings-type
+            :required:
+
+        ..  confval:: default
+            :type: mixed
+            :name: site-settings-definition-settings-default
+            :required:
```
Comment (froemken) [inline Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst:94]: You add a column "required" and marked "default" as "required", but this info was not rendered in column above

File: Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst [hunk 4]
```diff
@@ -8,88 +8,187 @@ Site settings definitions
...
 
-..  confval:: int
-    :name: site-setting-type-int
-    :type: string
-    :Path: settings.[my_val].type = int
+..  confval-menu::
+    :display: table
+    :name: site-setting-type
+    :type:
+    :required:
```
Comment (froemken) [inline Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst:116]: You add column "required", but not marked any following config as "required". Can be removed then, right?

File: Documentation/ApiOverview/SiteHandling/SiteSettingsEditor.rst [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+..  index:: Site handling; Site settings editor
+..  _site-settings-editor:
+
+====================
+Site settings editor
+====================
+
+..  versionadded:: 13.3
+    The site setting editor has been introduced as backend module
+    :guilabel:`Sites > Settings`.
```
Comment (froemken) [inline Documentation/ApiOverview/SiteHandling/SiteSettingsEditor.rst:11]: To be very correct it's "Site Management -> Settings"

File: Documentation/ApiOverview/SiteHandling/SiteSettingsEditor.rst [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+See the complete example at
+`settings.definitions.yaml (GitHub) <https://github.com/TYPO3-Documentation/blog_example/blob/main/Configuration/Sets/BlogExample/settings.definitions.yaml>`__.
+
+..  rst-class:: bignums-attention
+
+1.  Main category
+
+    The label of the category is defined in *line 3* of the example code snippet.
+    *Line 6 and 9* place two categories as subcategories into this category.
```
Comment (froemken) [inline Documentation/ApiOverview/SiteHandling/SiteSettingsEditor.rst:62]: Here and following you are working with line numbers, but as code box does not have any line numbers it's a little bit hard to reference them

File: Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst (hunk 4 above)
Comment (linawolf) [reply Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst:116]: This is displayed correctly when I render it: 
image

File: Documentation/ApiOverview/SiteHandling/SiteSettingsEditor.rst (hunk 6 above)
Comment (linawolf) [reply Documentation/ApiOverview/SiteHandling/SiteSettingsEditor.rst:63]: Good point, I display code lines

File: Documentation/ApiOverview/SiteHandling/SiteSettingsEditor.rst [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+current site, including their dependencies. The site sets can define categories
+and subcategories to order the settings.
+
+..  figure:: /Images/ManualScreenshots/SiteHandling/SiteSettings.png
+    :alt: Screenshot of the settings of an example site
+
+    The site in the examples includes the "My Sitepackage" and "Blog Example"
+    sets. "My Sitepackage" depends on "Fluid Styled Content"
+
+The settings to be displayed here have to be defined in an extension`s or
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/SiteHandling/SiteSettingsEditor.rst]: Done

File: Documentation/ApiOverview/SiteHandling/SiteSettingsEditor.rst (hunk 5 above)
Comment (linawolf) [reply Documentation/ApiOverview/SiteHandling/SiteSettingsEditor.rst]: good catch!

File: Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst (hunk 3 above)
Comment (linawolf) [reply Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst:94]: it is indeed required to have a default value.

File: Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst (hunk 2 above)
Comment (linawolf) [reply Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst]: Good catch!

File: Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/ApiOverview/SiteHandling/SiteSettingDefinitions.rst]: Rephrased it

**Comment by linawolf** [comment]
@froemken all open topics are resolved, can we merge this?

File: Documentation/ApiOverview/SiteHandling/SiteSettings.rst [hunk 8]
```diff
@@ -6,17 +6,11 @@
...
-    Settings are not removed from the :file:`config.yaml` file for now, but will
-    not have any effect anymore as soon as a :file:`settings.yaml` file exists.
-
-    Please review your settings in the :file:`config.yaml` file and remove them
-    manually. Eventually, you need and/or want to adopt your deployment
-    workflow.
+..  versionadded:: 13.1
+    Site settings can receive a type, a default value and some documentation in
+    :ref:`site settings definitions <site-sets-settings-definition>`. It is
+    recommended to always define a site setting before using is as only this way
```
Comment (Alagts) [inline Documentation/ApiOverview/SiteHandling/SiteSettings.rst:12]:
```suggestion
    recommended to always define a site setting before using it, as only this way
```

**Review by Alagts — APPROVED** [review]

**Pull request merged**